{
  "data": [
    {
      "_id": 1,
      "x": 11
    },
    {
      "_id": 2,
      "x": 22
    }
  ],
  "minServerVersion": "4.1.0",
  "tests": [
    {
      "description": "InsertOne succeeds after connection failure",
      "failPoint": {
        "configureFailPoint": "failCommand",
        "mode": {
          "times": 1
        },
        "data": {
          "failCommands": [
            "insert"
          ],
          "closeConnection": true
        }
      },
      "operation": {
        "name": "insertOne",
        "arguments": {
          "document": {
            "_id": 3,
            "x": 33
          }
        }
      },
      "outcome": {
        "result": {
          "insertedId": 3
        },
        "collection": {
          "data": [
            {
              "_id": 1,
              "x": 11
            },
            {
              "_id": 2,
              "x": 22
            },
            {
              "_id": 3,
              "x": 33
            }
          ]
        }
      }
    },
    {
      "description": "InsertOne succeeds after NotMaster",
      "failPoint": {
        "configureFailPoint": "failCommand",
        "mode": {
          "times": 1
        },
        "data": {
          "failCommands": [
            "insert"
          ],
          "errorCode": 10107,
          "closeConnection": false
        }
      },
      "operation": {
        "name": "insertOne",
        "arguments": {
          "document": {
            "_id": 3,
            "x": 33
          }
        }
      },
      "outcome": {
        "result": {
          "insertedId": 3
        },
        "collection": {
          "data": [
            {
              "_id": 1,
              "x": 11
            },
            {
              "_id": 2,
              "x": 22
            },
            {
              "_id": 3,
              "x": 33
            }
          ]
        }
      }
    },
    {
      "description": "InsertOne succeeds after NotMasterOrSecondary",
      "failPoint": {
        "configureFailPoint": "failCommand",
        "mode": {
          "times": 1
        },
        "data": {
          "failCommands": [
            "insert"
          ],
          "errorCode": 13436,
          "closeConnection": false
        }
      },
      "operation": {
        "name": "insertOne",
        "arguments": {
          "document": {
            "_id": 3,
            "x": 33
          }
        }
      },
      "outcome": {
        "result": {
          "insertedId": 3
        },
        "collection": {
          "data": [
            {
              "_id": 1,
              "x": 11
            },
            {
              "_id": 2,
              "x": 22
            },
            {
              "_id": 3,
              "x": 33
            }
          ]
        }
      }
    },
    {
      "description": "InsertOne succeeds after NotMasterNoSlaveOk",
      "failPoint": {
        "configureFailPoint": "failCommand",
        "mode": {
          "times": 1
        },
        "data": {
          "failCommands": [
            "insert"
          ],
          "errorCode": 13435,
          "closeConnection": false
        }
      },
      "operation": {
        "name": "insertOne",
        "arguments": {
          "document": {
            "_id": 3,
            "x": 33
          }
        }
      },
      "outcome": {
        "result": {
          "insertedId": 3
        },
        "collection": {
          "data": [
            {
              "_id": 1,
              "x": 11
            },
            {
              "_id": 2,
              "x": 22
            },
            {
              "_id": 3,
              "x": 33
            }
          ]
        }
      }
    },
    {
      "description": "InsertOne succeeds after InterruptedDueToReplStateChange",
      "failPoint": {
        "configureFailPoint": "failCommand",
        "mode": {
          "times": 1
        },
        "data": {
          "failCommands": [
            "insert"
          ],
          "errorCode": 11602,
          "closeConnection": false
        }
      },
      "operation": {
        "name": "insertOne",
        "arguments": {
          "document": {
            "_id": 3,
            "x": 33
          }
        }
      },
      "outcome": {
        "result": {
          "insertedId": 3
        },
        "collection": {
          "data": [
            {
              "_id": 1,
              "x": 11
            },
            {
              "_id": 2,
              "x": 22
            },
            {
              "_id": 3,
              "x": 33
            }
          ]
        }
      }
    },
    {
      "description": "InsertOne succeeds after InterruptedAtShutdown",
      "failPoint": {
        "configureFailPoint": "failCommand",
        "mode": {
          "times": 1
        },
        "data": {
          "failCommands": [
            "insert"
          ],
          "errorCode": 11600,
          "closeConnection": false
        }
      },
      "operation": {
        "name": "insertOne",
        "arguments": {
          "document": {
            "_id": 3,
            "x": 33
          }
        }
      },
      "outcome": {
        "result": {
          "insertedId": 3
        },
        "collection": {
          "data": [
            {
              "_id": 1,
              "x": 11
            },
            {
              "_id": 2,
              "x": 22
            },
            {
              "_id": 3,
              "x": 33
            }
          ]
        }
      }
    },
    {
      "description": "InsertOne succeeds after PrimarySteppedDown",
      "failPoint": {
        "configureFailPoint": "failCommand",
        "mode": {
          "times": 1
        },
        "data": {
          "failCommands": [
            "insert"
          ],
          "errorCode": 189,
          "closeConnection": false
        }
      },
      "operation": {
        "name": "insertOne",
        "arguments": {
          "document": {
            "_id": 3,
            "x": 33
          }
        }
      },
      "outcome": {
        "result": {
          "insertedId": 3
        },
        "collection": {
          "data": [
            {
              "_id": 1,
              "x": 11
            },
            {
              "_id": 2,
              "x": 22
            },
            {
              "_id": 3,
              "x": 33
            }
          ]
        }
      }
    },
    {
      "description": "InsertOne succeeds after ShutdownInProgress",
      "failPoint": {
        "configureFailPoint": "failCommand",
        "mode": {
          "times": 1
        },
        "data": {
          "failCommands": [
            "insert"
          ],
          "errorCode": 91,
          "closeConnection": false
        }
      },
      "operation": {
        "name": "insertOne",
        "arguments": {
          "document": {
            "_id": 3,
            "x": 33
          }
        }
      },
      "outcome": {
        "result": {
          "insertedId": 3
        },
        "collection": {
          "data": [
            {
              "_id": 1,
              "x": 11
            },
            {
              "_id": 2,
              "x": 22
            },
            {
              "_id": 3,
              "x": 33
            }
          ]
        }
      }
    },
    {
      "description": "InsertOne succeeds after HostNotFound",
      "failPoint": {
        "configureFailPoint": "failCommand",
        "mode": {
          "times": 1
        },
        "data": {
          "failCommands": [
            "insert"
          ],
          "errorCode": 7,
          "closeConnection": false
        }
      },
      "operation": {
        "name": "insertOne",
        "arguments": {
          "document": {
            "_id": 3,
            "x": 33
          }
        }
      },
      "outcome": {
        "result": {
          "insertedId": 3
        },
        "collection": {
          "data": [
            {
              "_id": 1,
              "x": 11
            },
            {
              "_id": 2,
              "x": 22
            },
            {
              "_id": 3,
              "x": 33
            }
          ]
        }
      }
    },
    {
      "description": "InsertOne succeeds after HostUnreachable",
      "failPoint": {
        "configureFailPoint": "failCommand",
        "mode": {
          "times": 1
        },
        "data": {
          "failCommands": [
            "insert"
          ],
          "errorCode": 6,
          "closeConnection": false
        }
      },
      "operation": {
        "name": "insertOne",
        "arguments": {
          "document": {
            "_id": 3,
            "x": 33
          }
        }
      },
      "outcome": {
        "result": {
          "insertedId": 3
        },
        "collection": {
          "data": [
            {
              "_id": 1,
              "x": 11
            },
            {
              "_id": 2,
              "x": 22
            },
            {
              "_id": 3,
              "x": 33
            }
          ]
        }
      }
    },
    {
      "description": "InsertOne succeeds after SocketException",
      "failPoint": {
        "configureFailPoint": "failCommand",
        "mode": {
          "times": 1
        },
        "data": {
          "failCommands": [
            "insert"
          ],
          "errorCode": 9001,
          "closeConnection": false
        }
      },
      "operation": {
        "name": "insertOne",
        "arguments": {
          "document": {
            "_id": 3,
            "x": 33
          }
        }
      },
      "outcome": {
        "result": {
          "insertedId": 3
        },
        "collection": {
          "data": [
            {
              "_id": 1,
              "x": 11
            },
            {
              "_id": 2,
              "x": 22
            },
            {
              "_id": 3,
              "x": 33
            }
          ]
        }
      }
    },
    {
      "description": "InsertOne succeeds after NetworkTimeout",
      "failPoint": {
        "configureFailPoint": "failCommand",
        "mode": {
          "times": 1
        },
        "data": {
          "failCommands": [
            "insert"
          ],
          "errorCode": 89,
          "closeConnection": false
        }
      },
      "operation": {
        "name": "insertOne",
        "arguments": {
          "document": {
            "_id": 3,
            "x": 33
          }
        }
      },
      "outcome": {
        "result": {
          "insertedId": 3
        },
        "collection": {
          "data": [
            {
              "_id": 1,
              "x": 11
            },
            {
              "_id": 2,
              "x": 22
            },
            {
              "_id": 3,
              "x": 33
            }
          ]
        }
      }
    },
    {
      "description": "InsertOne fails after Interrupted",
      "failPoint": {
        "configureFailPoint": "failCommand",
        "mode": {
          "times": 1
        },
        "data": {
          "failCommands": [
            "insert"
          ],
          "errorCode": 11601,
          "closeConnection": false
        }
      },
      "operation": {
        "name": "insertOne",
        "arguments": {
          "document": {
            "_id": 3,
            "x": 33
          }
        }
      },
      "outcome": {
        "error": true,
        "collection": {
          "data": [
            {
              "_id": 1,
              "x": 11
            },
            {
              "_id": 2,
              "x": 22
            }
          ]
        }
      }
    }
  ]
}
